### DISCUSSION TITLE
How can I debug wasm on Flutter Web

### DISCUSSION THREAD

**Discussion by idick7069** [simolus3/drift#2505] [Q&A]
I am try to build a flutter web project by using wasm on vscode
My code as same as the sample code here 
[https://drift.simonbinder.eu/web/#getting-started](url)

But i don't know how can I debug that 
I try to open wasm file with DB Browser for sqlite

But that need password?
![截圖 2023-07-01 下午3 33 14](https://github.com/simolus3/drift/assets/21211686/44635881-37eb-4501-9df1-9ce20ce8bf99)

And then I want to see log by using `logStatements`
But Database is not NativeDatabase. 

Can anyone help me please?

**Comment by simolus3** [comment]
The `sqlite3.wasm` file is not a database, it is the compiled sqlite3 library (so just a bunch of bytecode).

Unfortunately, at the moment it's not easy to log statements made to the database (because statement logging in drift is implemented at a fairly low level which in this case might be hosted in a web worker). If the chosen implementation is based on OPFS, you can use a plugin like [OPFS explorer](https://chrome.google.com/webstore/detail/opfs-explorer/acndjpgkpaclldomagafnognkcgjignd) to download the database. For IndexedDb-based databases, you can get the database file like this:

```dart
Future<Uint8List> loadDatabase() async {
  final fs = await IndexedDbFileSystem.open(dbName: 'your_database_name');
  final (file: file, outFlags: _) = fs.xOpen(Sqlite3Filename('/database'), 0);
  final blob = Uint8List(file.xFileSize());
  file.xRead(blob, 0);
  file.xClose();

  return blob;
}
```
